### PR TITLE
fix: Add test in GitHub Actions

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: cegao <ce.gao@outlook.com>

### CODE REVIEW

**Comment by Triple-Z** [comment]
TBD: `test-alipay` and `test-wechat` don't have assertions, consider to implement it.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    env:
+      GOPATH: ${{ github.workspace }}/go
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/deb-sig/double-entry-generator
+   steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/kubeflow/training-operator
```
Comment (Triple-Z) [inline .github/workflows/build.yml:18]: `${{ env.GOPATH }}/src/github.com/kubeflow/training-operator` -> `${{ env.GOPATH }}/src/github.com/deb-sig/double-entry-generator`

File: Makefile [hunk 2]
```diff
@@ -67,7 +67,10 @@ clean: ## Clean all the temporary files
 	@rm -rf ./bin
 	@rm -rf ./test/output
 
-test: test-alipay test-wechat ## Run all tests
+test: test-alipay test-wechat test-go ## Run all tests
```
Comment (Triple-Z) [inline Makefile:70]: `test-go` can be the first task.

```makefile
test: test-go test-alipay test-wechat
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          go-version: 1.17.2
+
+      - name: Check Go modules
+        run: |
+          go mod tidy && git add go.*
+          git diff --cached --exit-code || (echo 'Please run "go mod tidy" to sync Go modules' && exit 1);
+      - name: Verify gofmt
+        run: |
+          make format && git add pkg cmd
```
Comment (Triple-Z) [inline .github/workflows/build.yml]: This repo don't have `cmd` directory under the project root, try this instead:
```
make format && git add pkg
```

File: Makefile [hunk 4]
```diff
@@ -80,3 +80,6 @@ test-wechat: ## Run tests for WeChat provider
 
 format: ## Format code
 	@gofmt -s -w pkg
+
+check-format: ## Check if the format looks good.
+	@go fmt ./...
```
Comment (Triple-Z) [inline Makefile:85]: Specify the `pkg` directory is a better choice, or it will check the Go file format under the `vendor` directory.

Comment (gaocegege) [reply Makefile:85]: Should we remove vendor? It is not recommended to keep it in the repo.

Comment (gaocegege) [reply Makefile:85]: It ignores vendor by default, I think.

Comment (Triple-Z) [reply Makefile:85]: I ran the `gofmt -s -w .` yesterday, the `vendor` is included.

> Should we remove vendor? It is not recommended to keep it in the repo.

Yeah, sure.

**Comment by Triple-Z** [comment]
@gaocegege 

https://github.com/deb-sig/double-entry-generator/blob/442ccc6526e1ee458f9e838e8b56236863c4e112/pkg/provider/wechat/convert.go#L76

use `time.Format` in here.

format type: `2020-06-09 02:57:13 +0800 CST`

File: Makefile [hunk 5]
```diff
@@ -67,7 +67,10 @@ clean: ## Clean all the temporary files
...
 	@rm -rf ./test/output
 
-test: test-alipay test-wechat ## Run all tests
+test: test-go test-alipay test-wechat ## Run all tests
+
+test-go: ## Run Golang tests
+	@go test ./...
 
 test-alipay: ## Run tests for Alipay provider
 	@$(SHELL) ./test/alipay-test.sh
```
Comment (Triple-Z) [inline Makefile:76]: Can we add `[PASS]` assertion in these test scripts?

If the stdout doesn't have the `[PASS]`, then exit with 1 (error).

Maybe like this:
```bash
./test/alipay-test.sh | grep "[PASS]" > /dev/null || exit 1
```

Other providers are the same.

Comment (Triple-Z) [reply Makefile:76]: never mind.

Comment (gaocegege) [reply Makefile:76]: I am not sure why we need it. The process will return 1 if there are test failures.

**Review by Triple-Z — CHANGES_REQUESTED** [review]
lgtm

**Review by Triple-Z — APPROVED** [review]

**Pull request merged**